src/Host/Client/Impl/Session/RSessionProvider.cs

```diff
@@ -194,33 +197,50 @@ public class RSessionProvider : IRSessionProvider {
                 }
 
                 OnBrokerStateChanged(new HostLoad());
```

Comment:
When we are only testing connection but not actually switching - it seems like 'broker changed' should not be firing?